configmap-reload.go

```diff
@@ -67,6 +76,109 @@ func init() {
 	prometheus.MustRegister(requestsByStatusCode)
 }
 
+// generateNewConf returns a fully rendered configuration file with support
+// for golang templating
+func generateNewConf(templatePath string) (string, error) {
+	rel, err := filepath.Rel(volumeDirs[0], templatePath)
+	if err != nil {
+		return "", err
+	}
+
+	// if it's not the template file we want, then we just return the raw contents
+	// this also doubles as not running without a template file being set.
+	if rel != *templateFile {
+		b, err := ioutil.ReadFile(templatePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to read file: %v", err)
+		}
+		return string(b), nil
+	}
+
+	log.Printf("rendering file '%s'", templatePath)
+
+	t := template.New("conf")
+
+	b, err := ioutil.ReadFile(templatePath)
```

Comment:
No, since the path being provided to it is absolute. If we set this to rel it will break, the `volumeMounts` is the "relative" directory we care about.